doc/governance.rst

```diff
@@ -0,0 +1,126 @@
+Scikit-learn governance and decision-making
+===========================================
+
+The purpose of this document is to formalize the governance process used by the
+scikit-learn project, to clarify how decisions are made and how the various
+elements of our community interact.
+This document establishes a decision-making structure that takes into account
+feedback from all members of the community and strives to find consensus, while
+avoiding any deadlocks.
+
+This is a meritocratic, consensus-based community project. Anyone with an
+interest in the project can join the community, contribute to the project
+design and participate in the decision making process. This document describes
+how that participation takes place and how to set about earning merit within
+the project community.
+
+Roles And Responsibilities
+==========================
+
+Contributors
+------------
+Contributors are community members who contribute in concrete ways to the
+project. Anyone can become a contributor, and contributions can take many forms
+– not only code – as detailed in the `contributors guide <contributing>`.
+
+Core developers
+---------------
+Core developers are community members who have shown that they are dedicated to
+the continued development of the project through ongoing engagement with the
+community. Being a core developer  allows contributors to more easily carry on
+with their project related activities by giving them direct access to the
+project’s repository and is represented as being an organization member on the
+scikit-learn `GitHub organization <https://github.com/orgs/scikit-learn/people>`_.
+Core developers are expected to review code
+contributions, can merge approved pull requests, can cast votes for and against
+merging a pull-request, and can be involved in deciding major changes to the
+API.
+
+New core developers can be nominated by any existing core developers. Once they
+have been nominated, there will be a vote by the current core developers.
+Voting on new core developers is one of the few activities that takes place on
+the project's private management list. While it is expected that most votes
+will be unanimous, a two-thirds majority of the cast votes is enough. The vote
+needs to be open for at least 1 week.
+
+Core developers that have not contributed to the project (commits or GitHub
+comments) in the past 12 months will be asked if they want to become emeritus
+core developers and recant their commit and voting rights until they become
+active again. The list of core developers, active and emeritus (with dates at
+which they became active) is public on the scikit-learn website.
+
+Technical Committee
+-------------------
+The Technical Committee (TC) members are core developers who have additional
+responsibilities to ensure the smooth running of the project. TC members are expected to
+participate in strategic planning, and approve changes to the governance model.
+
```

Comment:
While discussing this document with various person, inside and outside the scikit-learn community, several asked why a TC was useful, rather than a flat hierarchy. I feel that this is a legitimate question, and maybe we could add the following text:
```suggestion

The purpose of the TC is to speed up big-picture decision and avoid loss of information.
Indeed, changes that impact the full project require a synthetic analysis and an consensus
that is both explicit and informed. Asking for an opinion from the full set of core
developers can easily lead to stale decision processes. The TC is a smaller group than
the full set of core developers, consulted only seldom, but that has the duty of building
informed decisions. 
```

In a sense, I would like the governance document to stay short (which is an argument to not add this text). However, I think that explaining why there is a TC is important, to avoid giving wrong impressions.

Reply:
I don't have strong feelings for or against adding this (i.e. +0)

Reply:
According to the proposed "consensus seeking" process outlined under the "Decision Making Process", as long as there's a clear consensus among the core developers, the TC doesn't have to interfere much, therefore the statement "The TC members ... approve changes to the governance model" is inaccurate here, and so is Gael's version. Maybe something like the following?:

The purpose of the TC is to ensure a smooth progress from the big-picture perspective.
Indeed changes that impact the full project require a synthetic analysis and a consensus
that is both explicit and informed. In cases that the core developer community
(which includes the TC members) fails to reach such a consensus in the required
time frame, the TC is the entity to resolve the issue.

Reply:
Wanna suggest an edit?

Reply:
```suggestion
The purpose of the TC is to ensure a smooth progress from the big-picture
perspective. Indeed changes that impact the full project require a synthetic
analysis and a consensus that is both explicit and informed. In cases that the
core developer community (which includes the TC members) fails to reach such a
consensus in the required time frame, the TC is the entity to resolve the
issue.
```